test_vm/src/lib.rs

```diff
@@ -613,6 +615,14 @@ impl<'invocation, 'bs> InvocationCtx<'invocation, 'bs> {
             }
         };
 
+        // But only if we're not in read-only mode.
+        if self.read_only() {
+            return Err(ActorError::unchecked(
+                ExitCode::USR_READ_ONLY,
+                format!("cannot create actor {target} in read-only mode"),
+            ));
+        }
```

Comment:
Can you point to the spec for read-only? Depending on what it says, I would expect either:
- no checks like this, but something at the end of an invocation that restores the state root to its pre-invocation value, or
- a lot more checks in `create`, `transaction`, `new_actor_address`, `create_actor`, delete_actor` (or something lower down that propagates through these) that actively prevent modifications.

Reply:
Spec: https://github.com/filecoin-project/FIPs/blob/322604f79c2447a4906ed2fc959e43aaedeccb46/FIPS/fip-0054.md#changed-syscalls

You're right. I've added checks to `create`, `create_actor`, `transaction`, `set_state_root` and `invoke`. 

I'm tweaking the behaviour of the test_vm's `new_actor_address` to be read-only, which is how it's actually implemented. The increment in moving to `create_actor` which is where it belongs.